Content search result summary

tubbo opened the pull request:
It's not always desirable for a content block to have it's data indexed. Any value in the `data` hash that appears more than 5 words is indexed and used as the summary during a content search. Example system test to demonstrate the issue:

```
      def test_content_search_yields_weird_results
        update_search_settings
        Content.for(create_page(name: 'Content Test')).tap do |content|
          content.blocks.create!(
            type: :hero,
            data: { asset: 'required', style: 'this is a test string'}
          )
        end
        visit storefront.search_path(q: 'content')
        assert(page.has_content?('this is a test string'))
      end
```

bencrouse commented:
Thanks @ryaan-anthony, did you have specific thoughts on how to control this? Is it an option at the content level, or content block level maybe?

ryaan-anthony commented:
We opted to use the content meta_description since it is technically for search/results, and we fallback on an allow list of block types (the client also doesn't want to show image `alt_text` in the summary). I honestly think the content block field should have a searchable flag on it. That would give the most control.

bencrouse commented:
Field level seems like it could be a bit much UI-wise, but I like the idea of using meta descriptions instead - it would allow opting out pretty easy by just leaving it blank. I’ll slot this in for an upcoming release, thanks.

Commit:
Prefer Meta Description for Search Results Text

If a `meta_description` is filled out, use that text instead of the
collected text from all content blocks.

bencrouse commented on app/models/workarea/search/storefront/content.rb:
```diff
@@ -27,7 +27,7 @@ def name
 
         def text
-          @text ||= ExtractContentBlockText.new(model.blocks).text
+          @text ||= model.meta_description || text_extracted_from_blocks
```
Can we skip extracting text from blocks altogether? I think the root of the problem is the way we're doing that and removing it would also give them a way to exclude things from results. We can chat about this.

tubbo replied:
I think we should do that in a future minor, but for now we should probably let it fall back to the original behavior to avoid a breaking change.

meowsus commented:
@tubbo these changes seemed slated for the next minor but not necessarily dependent on Workarea v3.5.0. I put it in the Major/Minor Release Candidates project. Let me know if you agree with that move, or if I missed its dependency on Workarea's next minor version.

bencrouse commented:
We're going to hold off until v3.6 and do a major release of this along with a few other improvements.
